client/lxd_images.go

```diff
@@ -608,6 +608,115 @@ func (r *ProtocolLXD) CopyImage(source ImageServer, image api.Image, args *Image
 		return nil, err
 	}
 
+	// Relay mode
+	if args != nil && args.Mode == "relay" {
+		metaFile, err := ioutil.TempFile("", "lxc_image_")
+		if err != nil {
+			return nil, err
+		}
+		defer os.Remove(metaFile.Name())
+
+		rootfsFile, err := ioutil.TempFile("", "lxc_image_")
+		if err != nil {
+			return nil, err
+		}
+		defer os.Remove(rootfsFile.Name())
+
+		// Import image
+		req := ImageFileRequest{
+			MetaFile:   metaFile,
+			RootfsFile: rootfsFile,
+		}
+
+		_, err = source.GetImageFile(image.Fingerprint, req)
+		if err != nil {
+			return nil, err
+		}
+
+		// Export image
+		_, err = metaFile.Seek(0, 0)
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = rootfsFile.Seek(0, 0)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
This appears to expect a split image (two files) and not support the unified image case (single file) case.

It shouldn't be too hard to look at what we received and then adjust the `createArgs` to match.

Reply:
This seems to work just fine with unified images.

Reply:
Hmm, that's odd, I would have expected that setting up `createArgs` later with both `MetaFile` and `RootfsFile` set would get us into the codepath where we use a multipart transfer which should be reserved to split images.

For unified, I would have expected us to make a `createArgs` with only `MetaFile` and `MetaName` set.

Reply:
I still don't like this part ^